src/renderer/base/renderer.hpp

```diff
@@ -45,114 +45,66 @@ namespace Microsoft::Console::Render
-
-        void _PaintBufferOutputGridLineHelper(_In_ IRenderEngine* const pEngine,
-                                              const TextAttribute textAttribute,
-                                              const size_t cchLine,
-                                              const COORD coordTarget);
-
+        void _PaintBufferOutputGridLineHelper(_In_ IRenderEngine* const pEngine, const TextAttribute textAttribute, const size_t cchLine, const COORD coordTarget);
         void _PaintSelection(_In_ IRenderEngine* const pEngine);
         void _PaintCursor(_In_ IRenderEngine* const pEngine);
-
         void _PaintOverlays(_In_ IRenderEngine* const pEngine);
         void _PaintOverlay(IRenderEngine& engine, const RenderOverlay& overlay);
-
-        [[nodiscard]] HRESULT _UpdateDrawingBrushes(_In_ IRenderEngine* const pEngine,
-                                                    const TextAttribute attr,
-                                                    const bool usingSoftFont,
-                                                    const bool isSettingDefaultBrushes);
-
+        [[nodiscard]] HRESULT _UpdateDrawingBrushes(_In_ IRenderEngine* const pEngine, const TextAttribute attr, const bool usingSoftFont, const bool isSettingDefaultBrushes);
         [[nodiscard]] HRESULT _PerformScrolling(_In_ IRenderEngine* const pEngine);
-
-        Microsoft::Console::Types::Viewport _viewport;
-
-        static constexpr float _shrinkThreshold = 0.8f;
-        std::vector<Cluster> _clusterBuffer;
-
         std::vector<SMALL_RECT> _GetSelectionRects() const;
         void _ScrollPreviousSelection(const til::point delta);
-        std::vector<SMALL_RECT> _previousSelection;
-
         [[nodiscard]] HRESULT _PaintTitle(IRenderEngine* const pEngine);
-
         [[nodiscard]] std::optional<CursorOptions> _GetCursorInfo();
         [[nodiscard]] HRESULT _PrepareRenderInfo(_In_ IRenderEngine* const pEngine);
+        bool _isSoftFontChar(const std::wstring_view& v) const noexcept;
 
-        const size_t _firstSoftFontChar = 0xEF20;
+        std::array<IRenderEngine*, 2> _engines{};
+        IRenderData* _pData = nullptr; // Non-ownership pointer
+        std::unique_ptr<IRenderThread> _pThread;
+        static constexpr size_t _firstSoftFontChar = 0xEF20;
         size_t _lastSoftFontChar = 0;
-        static bool s_IsSoftFontChar(const std::wstring_view& v, const size_t firstSoftFontChar, const size_t lastSoftFontChar);
-
-        // Helper functions to diagnose issues with painting and layout.
-        // These are only actually effective/on in Debug builds when the flag is set using an attached debugger.
-        bool _fDebug = false;
-
+        std::optional<interval_tree::IntervalTree<til::point, size_t>::interval> _hoveredInterval;
+        Microsoft::Console::Types::Viewport _viewport;
+        std::vector<Cluster> _clusterBuffer;
+        std::vector<SMALL_RECT> _previousSelection;

```

Comment:
I've moved all member functions up and all member variables down.
I didn't pay any attention to ordering of the functions but I tried packing the variables a bit.